hibernate-core/src/main/java/org/hibernate/query/sqm/tree/predicate/SqmBooleanExpressionPredicate.java

```diff
@@ -73,7 +72,7 @@ public <T> T accept(SemanticQueryWalker<T> walker) {
 
 	@Override
 	public List<Expression<Boolean>> getExpressions() {
-		return Collections.singletonList( booleanExpression );
+		return Arrays.asList( booleanExpression );
```

Comment:
The list which `Arrays.asList` returns can not be added to unfortunately, so we need to use this:

```suggestion
		final List<Expression<Boolean>> list = new ArrayList<>( 1 );
		list.add( booleanExpression );
		return list;
```

Reply:
Thanks @beikov